### DISCUSSION TITLE
Image updates only work locally but not on remote Ubuntu server.

### DISCUSSION THREAD

**Discussion by tdao-usesi** [containrrr/watchtower#1229] [Q&A]
I have watchtower running on both my local machine as well as my Ubuntu server. When I push a new image onto Docker Hub with updates to my code, the updated image is detected and a new container is spun up locally with the most recent image. On my remote server, watchtower detects the new image and also spins up a new container, but this container does not include my new changes.

Here are my server logs:
![image](https://user-images.githubusercontent.com/91479415/154101293-5bf377a4-6bd5-4d3d-9700-d3d1a2d638fe.png)

Here are my local logs:
![image](https://user-images.githubusercontent.com/91479415/154101616-517dd139-e591-416a-9395-6c4c9772c2e5.png)

**Comment by piksel** [comment]
> this container does not include my new changes  

How are you determining that?
You can check what the SHA256 hash of the current container image using:
```
docker inspect pawebapp | jq '.[].Image'
```
(or if you don't have `jq`, just look for the first `Image: ` in the output).

You cut your logs just before the new hash was logged, so I don't know what it should be, but it should match the one in the log.

**Reply by tdao-usesi** [reply]
I have a basic HTML page used to manually test if watchtower pulls the most recent image correctly. I just added a p tag saying "Test..." and used this command to get inside the containers.

`docker exec -it pawebapp sh`

The changes were not shown in the container running in the server, but they were there on my local container.

I'll attach my logs again with more included.
Server logs:
![image](https://user-images.githubusercontent.com/91479415/154513671-f1f17396-6b10-4659-995d-03a84e41096c.png)

Running your command on my server shows this:

`"Image": "sha256:c8df7a90b18e698d6c28fe768c9a9145b2f51e4a0bd0137692480443780be61d"`

Local logs:
![image](https://user-images.githubusercontent.com/91479415/154513799-0f364c4e-023f-4140-a850-5a410c0d4c38.png)

Locally, it shows this:

`"Image": "sha256:c8df7a90b18e698d6c28fe768c9a9145b2f51e4a0bd0137692480443780be61d"`

Seems like the correct image is being used, but not sure why the changes aren't showing up on my server.

**Comment by simonaronsson** [comment]
Would you mind posting the command(s) you are running to deploy watchtower?
